test/src/common/bridgePowerController_ut.cpp

```diff
@@ -466,12 +633,19 @@ TEST_F(BridgePowerControllerTest, goldenPath) {
   xTaskSetTickCount(0); // Convinience tick set for checking sleep.
   BridgePowerController.powerControlEnable(true);
   BridgePowerController._update();
-  EXPECT_EQ(fake_io_read_func_fake.call_count, 5);
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 4);
   EXPECT_EQ(fake_io_write_func_fake.call_count, 1);
 
+  // Controller waits until the next aligned sample start
+  uint32_t curtime = BridgePowerController::DEFAULT_SAMPLE_INTERVAL_S * 1000;
+  EXPECT_EQ(xTaskGetTickCount(), curtime);
+
   // The bus stays on until the next Sample Off time
-  uint32_t curtime =
-      (BridgePowerController::DEFAULT_SAMPLE_INTERVAL_S + SAMPLE_DURATION_S) * 1000;
+  rtcGetMicroSeconds_fake.return_val = curtime * 1000;
+  BridgePowerController._update();
+  EXPECT_EQ(fake_io_read_func_fake.call_count, 5);
+  EXPECT_EQ(fake_io_write_func_fake.call_count, 1);
+  curtime += SAMPLE_DURATION_S * 1000;
```

Comment:
The golden path test had to be slightly modified because it was written to rely on the behavior of sleeping past a sample start, which is the bug we're fixing. Other expectations remain the same, but there's an extra wake-up and so the test contains an extra call to `_update()`. I split the `curtime` expectations into 2 components, first up to the sample interval, then to the end of the sample duration.